bpf/lib/nat.h

```diff
@@ -746,19 +748,21 @@ snat_v4_needs_masquerade(struct __ctx_buff *ctx __maybe_unused,
 	 * skip the CT lookup since this cannot be reply traffic.
 	 */
 	if (local_ep) {
-		struct ipv4_ct_tuple tuple = {
-			.nexthdr = ip4->protocol,
-			.daddr = ip4->daddr,
-			.saddr = ip4->saddr
-		};
+		__u8 flags = tuple->flags;
 		int err;
 
 		target->from_local_endpoint = true;
 
-		err = ct_is_reply4(get_ct_map4(&tuple), ctx, ETH_HLEN +
-				   ipv4_hdrlen(ip4), &tuple, &is_reply);
+		err = ct_is_reply4(get_ct_map4(tuple), ctx, l4_off, tuple,
+				   &is_reply);
 		if (IS_ERR(err))
 			return err;
+
+		/* restore initial flags */
+		tuple->flags = flags;
+		/* SNAT code has its own port extraction logic: */
+		tuple->dport = 0;
+		tuple->sport = 0;
```

Comment:
maybe to wrap this into an inline function defined next to `ct_is_reply*` and comment that it should be in sync?